tests/test_hf_api.py

```diff
@@ -928,6 +929,31 @@ def test_model_info_with_security(self):
             {"containsInfected": False},
         )
 
+    @with_production_testing
+    def test_model_info_with_file_metadata(self):
+        _api = HfApi()
+        model = _api.model_info(
+            repo_id=DUMMY_MODEL_ID,
+            revision=DUMMY_MODEL_ID_REVISION_ONE_SPECIFIC_COMMIT,
+            files_metadata=True,
+        )
+        files = model.siblings
+        assert files is not None
+        self.assertListEqual(
+            [isinstance(file.blob_id, str) for file in files], [True] * len(files)
+        )
+        self.assertListEqual(
+            [isinstance(file.size, int) for file in files], [True] * len(files)
+        )
+        self.assertTrue(any([file.lfs is not None for file in files]))
+        self.assertListEqual(
+            [
+                file.lfs is None or isinstance(file.lfs, dict) and "sha256" in file.lfs
+                for file in files
+            ],
+            [True] * len(files),
+        )
```

Comment:
Quick advice here (maybe more a personal opinion), I always try to favor code readability especially in the tests. Here I would rather loop over the file and assert stuff on their content instead of building artificial lists `[True] * len(files)` that slow down understanding.


```suggestion
        at_least_one_lfs = False
        for file in files:
            self.assertTrue(isinstance(file.blob_id, str))
            self.assertTrue(isinstance(file.size, int))
            if file.lfs is not None:
                at_least_one_lfs = True
                self.assertTrue(isinstance(file.lfs, dict) and "sha256" in file.lfs)
        self.assertTrue(at_least_one_lfs)
```